src/items.rs

```diff
@@ -113,17 +113,57 @@ impl Rewrite for ast::Local {
         result.push_str(&infix);
 
         if let Some((init, _els)) = self.kind.init_else_opt() {
+            let base_span = if let Some(ref ty) = self.ty {
+                mk_sp(ty.span.hi(), self.span.hi())
+            } else {
+                mk_sp(self.pat.span.hi(), self.span.hi())
+            };
+
+            let offset = context.snippet(base_span).find_uncommented("=")?;
+            let base_span_lo = base_span.lo();
+
+            let assign_lo = base_span_lo + BytePos(offset as u32);
+            let comment_start_pos = if let Some(ref ty) = self.ty {
+                ty.span.hi()
+            } else {
+                self.pat.span.hi()
+            };
+            let comment_before_assign = context.snippet(mk_sp(comment_start_pos, assign_lo)).trim();
+
+            let assign_hi = base_span_lo + BytePos((offset + 1) as u32);
+            let rhs_span_lo = init.span.lo();
+            let comment_end_pos = if init.attrs.is_empty() {
+                rhs_span_lo
+            } else {
+                let attr_span_lo = init.attrs.first().unwrap().span.lo();
+                // for the case using block
+                // ex. let x = { #![my_attr]do_something(); }
+                if rhs_span_lo < attr_span_lo {
+                    rhs_span_lo
+                } else {
+                    attr_span_lo
+                }
+            };
+
+            if !comment_before_assign.is_empty() {
+                let new_indent_str = &pat_shape
+                    .block_indent(0)
+                    .to_string_with_newline(context.config);
+                result = format!("{}{}{}", comment_before_assign, new_indent_str, result);
+            }
+
             // 1 = trailing semicolon;
             let nested_shape = shape.sub_width(1)?;
-
-            result = rewrite_assign_rhs(
+            result = rewrite_assign_rhs_with_comments(
                 context,
                 result,
                 init,
-                &RhsAssignKind::Expr(&init.kind, init.span),
                 nested_shape,
+                &RhsAssignKind::Expr(&init.kind, init.span),
+                RhsTactics::Default,
+                mk_sp(assign_hi, comment_end_pos),
+                true,
             )?;
-            // todo else
```

Comment:
Correct me if I'm wrong, but we don't yet handle the `else` case do we? I don't think it's okay to remove this TODO comment if we haven't handled that case yet.

Reply:
The TODO removal was done because it was removed by this [R2 commit](https://github.com/davidBar-On/rustfmt/commit/f897e25b0fbb146da7fffd3b06fe9a9c785e8ecb#), which was the basis for the changes.

I checked and it seems that the removal of the TODO is o.k.  [`init_else_opt()`](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_ast/ast.rs.html#1036-1042) handles both `Iniit` and `InitElse` kinds.  It is o.k. that`Decl` kind is not handled, as at this point the code that follows the assignment `=` is handled.